ert/data/record/_record.py

```diff
@@ -98,9 +121,29 @@ def __eq__(self, o: object) -> bool:
 
 
 class NumericalRecord(Record):
+    """The :class:`NumericalRecord` is implementation of the Record class
+    that handles numerical data.
+    """
+
     def __init__(
         self, data: numerical_record_data, index: Optional[RecordIndex] = None
     ) -> None:
+        """Creates a NumericalRecord instance.
+        It automatically assigns :class:`RecordType` as either
+        :class:`RecordType.LIST_FLOAT`, :class:`RecordType.MAPPING_INT_FLOAT` or
+        :class:`RecordType.MAPPING_STR_FLOAT`.
+        The data index is either automatically inferred from the data or
+        can be provided as a parameter.
+
+        Args:
+            data: numerics that cannot be None
+            index: data indices, which defaults to None.
```

Comment:
Maybe something like 
```suggestion
            index: data indices. If None is given, then index will be inferred from the data.
```